openstates/ak/bills.py

```diff
@@ -198,109 +205,159 @@ def scrape_bill(self, chamber, session, bill_id, bill_type, url):
             bill.add_subject(subj.strip())
 
-        # Get versions
+        # Get versions - to do
         text_list_url = (
-            "http://www.legis.state.ak.us/"
-            "basis/get_fulltext.asp?session=%s&bill=%s"
-        ) % (session, bill_id)
+            "https://www.akleg.gov/basis/Bill/Detail/%s?Root=%s#tab1_4") % (session, bill_id)
         bill.add_source(text_list_url)
 
         text_doc = lxml.html.fromstring(self.get(text_list_url).text)
         text_doc.make_links_absolute(text_list_url)
-        for link in text_doc.xpath('//a[contains(@href, "get_bill_text")]'):
-            name = link.xpath('../preceding-sibling::td/text()')[0].strip()
+        for link in text_doc.xpath('//a[contains(@href, "/Text/")]'):
+            name = link.text_content()
             text_url = link.get('href')
             bill.add_version_link(name, text_url, media_type="text/html")
 
-        # Get documents
+        # Get documents - to do
         doc_list_url = (
-            "http://www.legis.state.ak.us/"
-            "basis/get_documents.asp?session=%s&bill=%s"
-        ) % (session, bill_id)
+            "https://www.akleg.gov/basis/Bill/Detail/%s?Root=%s#tab5_4") % (session, bill_id)
         doc_list = lxml.html.fromstring(self.get(doc_list_url).text)
         doc_list.make_links_absolute(doc_list_url)
         bill.add_source(doc_list_url)
         for href in doc_list.xpath('//a[contains(@href, "get_documents")][@onclick]'):
             h_name = href.text_content()
             h_href = href.attrib['href']
             if h_name.strip():
-                bill.add_document_link(h_name, h_href)
+                try:
+                    bill.add_document_link(h_name, h_href)
+                except:
+                    print("Duplicate found")
 
         yield bill
 
-    def parse_vote(self, bill, action, act_chamber, act_date, url):
-        re_vote_text = re.compile(r'The question (?:being|to be reconsidered):\s*"(.*?\?)"', re.S)
-        re_header = re.compile(r'\d{2}-\d{2}-\d{4}\s{10,}\w{,20} Journal\s{10,}\d{,6}\s{,4}')
-
+    def parse_vote(self, bill, journal_entry_number, action, act_chamber, act_date, url):
+        
         html = self.get(url).text
         doc = lxml.html.fromstring(html)
+        yes = no = other = 0
+        vote_counts = action.split()
+        for vote_count in vote_counts:
+            if re.match(r'[\D][\d]', vote_count):
+                if "Y" in vote_count:
+                    yes = int(vote_count[1:])
+                elif "N" in vote_count:
+                    no = int(vote_count[1:])
+                elif "E" in vote_count or "A" in vote_count:
+                    other += int(vote_count[1:])
+
+
+        vote = VoteEvent(
+                bill=bill,
+                start_date=act_date.strftime('%Y-%m-%d'),
+                chamber=act_chamber,
+                motion_text=action,
+                result='pass' if yes > no else 'fail',
```

Comment:
is this our best proxy for passage?  it is often incorrect so let's double check that there's no passage indicator on the page

Reply:
Fixed. Now checking action to see if passed or failed.